source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -468,6 +488,35 @@ TEST_F(NiDAQmxDriverApiTests, GetErrorString_ReturnsErrorMessage)
   EXPECT_SUCCESS(status, response);
   EXPECT_THAT(response.error_string(), HasSubstr("Requested value is not a supported value for this property."));
 }
+
+TEST_F(NiDAQmxDriverApiTests, ReadBinaryI32_Succeeds)
+{
+  create_ai_voltage_chan(-5.0, 5.0);
+  start_task();
+
+  ReadBinaryI32Response response;
+  const int32 NUM_SAMPS = 4;
+  auto status = read_binary_i32(NUM_SAMPS, response);
+  stop_task();
+
+  EXPECT_EQ(0, status.error_code());
+  EXPECT_SUCCESS(status, response);
+  EXPECT_EQ(NUM_SAMPS, response.samps_per_chan_read());
+}
+
+TEST_F(NiDAQmxDriverApiTests, AIVoltageChannel_CfgSampClkTimingAndAcquireData_Succeeds)
+{
+  create_ai_voltage_chan(0.0, 1.0);
+
+  CfgSampClkTimingResponse response;
+  auto config_status = cfg_samp_clk_timing(response);
+  start_task();
+  ReadAnalogF64Response read_response;
+  auto read_status = read_analog_f64(10, 10, read_response);
+
+  EXPECT_SUCCESS(config_status, response);
+  EXPECT_SUCCESS(read_status, read_response);
```

Comment:
Do we want to check the `samps_per_chan_read` on the `read_response` here as well?